clippy_lints/src/option_if_let_else.rs

```diff
@@ -129,6 +129,18 @@ fn detect_option_if_let_else<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'tcx>) ->
             let (as_ref, as_mut) = match &let_expr.kind {
                 ExprKind::AddrOf(_, Mutability::Not, _) => (true, false),
                 ExprKind::AddrOf(_, Mutability::Mut, _) => (false, true),
+                ExprKind::MethodCall(..) => {
+                    if let Some(fn_did) = fn_def_id(cx, let_expr) {
+                        let fn_sig = cx.tcx.fn_sig(fn_did).input(0);
+                        if let Some(Mutability::Mut) = cx.tcx.erase_late_bound_regions(fn_sig).ref_mutability() {
```

Comment:
This should be the same since you're not doing anything with regions.
```suggestion
                        if let Some(Mutability::Mut) = cx.tcx.fn_sig(fn_did).skip_binder().inputs()[0].ref_mutability() {
```